eos_data_distribution/ndn/base.py

```diff
@@ -114,6 +111,61 @@ def __init__(self, name, face=None):
 
         self._callbackCount = 0
         self._responseCount = 0
+        self._keyChain = None
+        self._certificateName = None
+        self.pit = dict()
+
+    def generateKeys(self):
+        # Use the system default key chain and certificate name to sign commands.
+        keyChain = KeyChain()
+        self._keyChain = keyChain
+        try:
+            self._certificateName = keyChain.getDefaultCertificateName()
+        except:
+            name = Name(self.name)
+            logger.warning("Could not get default certificate name, creating a new one from %s", name)
+            self._certificateName = keyChain.createIdentityAndCertificate(name)
+        self._responseCount = 0
+
+        self.face.setCommandSigningInfo(keyChain, self._certificateName)
+
+    def sign(self, data):
+        return self._keyChain.sign(data, self._certificateName)
+
+    def expressInterest(self, interest=None, forever=False, *args, **kwargs):
+        if interest is None:
+            interest = Interest(self.name)
+        return self._expressInterest (interest, self.name, *args, **kwargs)
+
+    def _expressInterest(self, interest, name=None,
+                         forever=False, onData=None, onTimeout=None):
+        if not name: name = self.name
+        if not onData: onData = self._onData
+        if not onTimeout: onTimeout = partial(self.onTimeout,
+                            forever=forever, name=name)
+
+        logger.debug("Express Interest name: %s", interest)
+        self.pit[interest] = self.face.expressInterest(interest, onData, onTimeout)
+        return interest
+
+    def makeCommandInterest(self, cmd, prefix=None, controlParameters=None,
+                            keyChain=None, certificateName=None,
+                            *args, **kwargs):
+        if not prefix: prefix=self.name
+        if not controlParameters: controlParameters = ControlParameters()
+        if not self._keyChain or not self._certificateName:
+            self.generateKeys()
```

Comment:
Do we need to call `self.generateKeys()` if we are not going to use `self._keyChain` and `self._certificateName` because the caller has provided `keyChain` and `certificateName`?